ably-ios/ARTAuth.m

```diff
@@ -274,6 +275,11 @@ - (void)setProtocolClientId:(NSString *)clientId {
 
 - (void)setTokenDetails:(ARTAuthTokenDetails *)tokenDetails {
     _tokenDetails = tokenDetails;
+    // If a token has been setted and the current Auth method is Basic then for now on use Token Authentication
```

Comment:
@mattheworiordan Just a way to guarantee the correct auth method is used when a `TokenDetails` but maybe it should be private.